File: conformance/tests/post-quantum-wots/post-quantum-wots.runar.py
from runar import SmartContract, ByteString, public, assert_, verify_wots

class PostQuantumWOTS(SmartContract):
    pubkey: ByteString

    def __init__(self, pubkey: ByteString):
        super().__init__(pubkey)
        self.pubkey = pubkey

    @public
    def spend(self, msg: ByteString, sig: ByteString):
        assert_(verify_wots(msg, sig, self.pubkey))

File: conformance/tests/post-quantum-wots/post-quantum-wots.runar.rb
require 'runar'

class PostQuantumWOTS < Runar::SmartContract
  prop :pubkey, ByteString

  def initialize(pubkey)
    super(pubkey)
    @pubkey = pubkey
  end

  runar_public msg: ByteString, sig: ByteString
  def spend(msg, sig)
    assert verify_wots(msg, sig, @pubkey)
  end
end
